doc/nrf/releases_and_maturity/migration/migration_guide_2.6.rst

```diff
@@ -652,6 +652,26 @@ The following changes are recommended for your application to work optimally aft
 
 * For LwM2M applications, replace the :kconfig:option:`CONFIG_LWM2M_CLIENT_UTILS_DTLS_CID` Kconfig option with :kconfig:option:`CONFIG_LWM2M_DTLS_CID`.
 
+* For Mesh applications, the Sensor API has been re-written, and the previous Sensor API is to be depreciated.
+  A new Sensor API has been created that should be used for new applications and applications that will be continuously updated.
+  How the new Sensor API is working is shown in the :ref:`bluetooth_mesh_sensor_client`, :ref:`bluetooth_mesh_sensor_server` and :ref:`bluetooth_mesh_light_lc` samples, as they all have been updated to use the new Sensor API.
+  It is important to use the new Sensor API as it has a better precision than the deprecated Sensor API.
+
+  The :kconfig:option:`CONFIG_BT_MESH_SENSOR_USE_LEGACY_SENSOR_VALUE` is set default to `=y`.
+  This means for existing applications there is no need to do anything to continue using the depreciated Sensor API.
+  The application will get a deprecation warning when compiled, that the user can choose to disregard.
+  This will work until the depreciation periode is over for the deprecated Sensor API.
+  When the depreciation period is over the deprecated Sensor API will be removed, and the application will no longer compile if not updated to the new Sensor API.
+
+  New and existing applications that shall use the new Sensor APIs must set :kconfig:option:`CONFIG_BT_MESH_SENSOR_USE_LEGACY_SENSOR_VALUE` to `=n` in their configuration.
+  This configuration value will be removed when the deprecation periode is over, and the configuration value then need to be removed from the application configuration.
+  It is important to check if :kconfig:option:`CONFIG_FPU` is set to `=y` to support floating point operations, and possible add :kconfig:option:`CONFIG_CBPRINTF_FP_SUPPORT` to support floating point printing.
+  For sensor functionality used in the application, application variables and function/callback variables defined as :c:struct:`sensor_value` shall now use :c:struct:`bt_mesh_sensor_value` instead.
+  The :c:struct:`bt_mesh_sensor_value`.format need to be defined for application variables.
+  There are several different types of :c:struct:`bt_mesh_sensor_format` described in the :file:`sensor_types.h`.
+  The :c:struct:`bt_mesh_sensor_format` has a :c:struct:`bt_mesh_sensor_format_cb` that is used through a set of converter functions for each format.
```

Comment:
This is an implementation detail and we should not mention it here, it will only cause people to start messing with `cb` directly which is something we specifically _don't_ want. 

Suggested rewording:
```suggestion
  :c:struct:`bt_mesh_sensor_value`s with a valid `format` can be converted to and from integer, float and `struct sensor_value` through `bt_mesh_sensor_value_to/from*` functions.
```